internal/proxyservice/proxy_service.go

```diff
@@ -57,89 +111,145 @@ func (s *proxyServices) Stop() error {
 	return nil
 }
 
-func (s *proxyServices) servicesToStart() []internal.Service {
-	var servicesToStart []internal.Service
-
+func (s *proxyServices) servicesToStart() (servicesToStart []internal.Service) {
+	// TODO: v2.NewConfigsByType should be an interface, so we can remove this
+	//   hardcoded dep on an impl type of v2.  All deps need to be injected.
+	configsByType := v2.NewConfigsByType(s.config.Services, s.availPlugins)
+	httpPlugins := s.availPlugins.HTTPPlugins()
 	tcpPlugins := s.availPlugins.TCPPlugins()
-	// httpPlugins := s.availPlugins.HTTPPlugins()
-
-	for _, svc := range s.config.Services {
-		requestedPlugin := svc.Connector //TODO: this rename is a name smell
-
-		// first check the available TCP Plugins
-		tcpPlugin, found := tcpPlugins[requestedPlugin]
-		if found {
-			tcpSvc, err := s.createTCPService(svc, tcpPlugin)
-			// Failure to create is fatal.
-			if err != nil {
-				// TODO: Add Fatalf to our logger and use that
-				s.logger.Panicf("unable to create TCP service '%s'", svc.Name)
-			}
-			servicesToStart = append(servicesToStart, tcpSvc)
-			continue
+
+	// TCP Plugins
+	for _, cfg := range configsByType.TCP {
+		// Validation will have already happened
+		tcpSvc, err := s.createTCPService(cfg, tcpPlugins[cfg.Connector])
+		if err != nil {
+			// TODO: Add Fatalf to our logger and use that
+			s.logger.Panicf("unable to create TCP service '%s'", cfg.Name)
```

Comment:
we should append the error to the panic message so the user has additional information as to the cause